### PR TITLE
New version: Microsoft.PowerShell version 7.3.8.0

### PR DESCRIPTION
**Pull request by anamnavi**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/122317)

### CODE REVIEW

File: manifests/m/Microsoft/PowerShell/7.3.8.0/Microsoft.PowerShell.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- silentWithProgress
+UpgradeBehavior: install
+Commands:
+- pwsh
+ReleaseDate: 2023-10-10
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/PowerShell/PowerShell/releases/download/v7.3.8/PowerShell-7.3.8-win-x64.msi
+  InstallerSha256: A6551A73522F24D62D03A36E6EB5AFD77098C975885D750A9E6E3CE06F8178C9
+  ProductCode: '{515EFDB5-CC15-4ADC-89A4-7B4D818D5ACD}'
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/PowerShell/7.3.8.0/Microsoft.PowerShell.installer.yaml:20]:
```suggestion
  ProductCode: '{1DE68B23-402E-4EA5-978D-5F1E9128AC2C}'
```

File: manifests/m/Microsoft/PowerShell/7.3.8.0/Microsoft.PowerShell.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- Architecture: x86
+  InstallerUrl: https://github.com/PowerShell/PowerShell/releases/download/v7.3.8/PowerShell-7.3.8-win-x86.msi
+  InstallerSha256: 5869F5F002F34B4985032D67E9B73FDDB0DD1013EAF992CFFC474AE687DFD071
+  ProductCode: '{09FBD65E-4956-45E6-B7C8-11EC7DB8A2F3}'
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/PowerShell/7.3.8.0/Microsoft.PowerShell.installer.yaml:24]:
```suggestion
  ProductCode: '{7FA35664-8619-4399-B863-689B152E00B6}'
```

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Comment by anamnavi** [comment]
thanks @mdanish-kh not sure why the product codes were off...I was doing it through my personal mac but thanks for the catch!

**Comment by mdanish-kh** [comment]
Since you used YamlCreate, I'm looking at these lines of code. Not sure if this flow works properly for a Mac and since I don't have one I can't verify 😄 

cc @Trenly if he wishes to take a stab at it

https://github.com/microsoft/winget-pkgs/blob/d6013c80c0c3915bf924772472925a3368895045/Tools/YamlCreate.ps1#L926-L929

**Review by mdanish-kh — APPROVED** [review]

**Comment by mdanish-kh** [comment]
@stephengillie - This needs a waiver for `Internal-Error-Dynamic-Scan` please

**Comment by mdanish-kh** [comment]
@stephengillie - This needs to be manually validated please

**Review by stephengillie — APPROVED** [review]

**Pull request merged**